docs/polymorfisme.md

```diff
@@ -0,0 +1,257 @@
+    datum:
+      type: string
+      format: date
+    jaar:
+      type: integer
+      maximum: 9999
+    maand:
+      type: integer
+      minimum: 1
+      maximum: 12
+    dag:
+      type: integer
+      minimum: 1
+      maximum: 31
+    type:
+      type: string
+      enum:
+        - Datum
+        - JaarMaandDatum
+        - JaarDatum
+        - OnbekendDatum
+  required:
+    - type
+```
+
+Het 'type' veld is gedefinieerd als een enumeratie om aan te kunnen geven wat de mogelijke waarden zijn. Ook is dit veld verplicht gemaakt om aan te geven dat dit veld altijd een waarde heeft.
+
+De logica voor het bepalen van de datum type kan dan als volgt worden herschreven:
+
+```javascript
+if(type === 'Datum') {
+    console.log("datum");
+}
+else if(type === 'JaarMaandDatum') {
+    console.log("dag onbekend");
+}
+else if(type === 'JaarDatum') {
+    console.log("maand en dag onbekend");
+}
+else if(type === 'OnbekendDatum') {
+    console.log("onbekend datum");
+}
+else {
+    console.log("niet ondersteund datum");
+}
+```
+
+Wat deze DatumOnvolledig definitie niet oplost, is dat het mogelijk is om een type aan te geven en niet alle bijhorende velden een waarde te geven.  
+
+Een manier om dit op te lossen is om voor elk datum type een eigen definitie te maken zodat expliciet kan worden aangegeven welke velden bij elk datum type horen. De vier datum types zien er dan als volgt uit:
+
+```yaml
+VolledigeDatum:
+  type: object
+  properties:
+    datum:
+      type: string
+      format: date
+    type:
+      type: string
+      enum:
+        - Datum
+  required:
+    - datum
+    - type
+JaarMaandDatum:
+  type: object
+  properties:
+    jaar:
+      type: integer
+      maximum: 9999
+    maand:
+      type: integer
+      minimum: 1
+      maximum: 12
+    type:
+      type: string
+      enum:
+        - JaarMaandDatum
+  required:
+    - jaar
+    - maand
+    - type
+JaarDatum:
+  type: object
+  properties:
+    jaar:
+      type: integer
+      maximum: 9999
+    type:
+      type: string
+      enum:
+        - JaarDatum
+  required:
+    - jaar
+    - type
+OnbekendDatum:
+  type: object
+  properties:
+    type:
+      type: string
+      enum:
+        - OnbekendDatum
+  required:
+    - type
+```
+
+Met deze vier datum definities is nu expliciet gedefinieerd welk datum type welke velden heeft en wat de bijbehorende waarde is in het 'type' veld. 
+
+Om de datum types te kunnen gebruiken als de mogelijke types van een datum veld moet [polymorfisme](https://nl.wikipedia.org/wiki/Polymorfisme) worden toegepast. In een OpenAPI Specificatie wordt [polymorfisme](https://spec.openapis.org/oas/v3.0.3#composition-and-inheritance-polymorphism) toegepast met behulp van de **allOf** en **discriminator** elementen:
+- er moet een basis type worden gedefinieerd voor de vier datum types. Deze basis type kan dan worden gebruikt als type voor een datum veld. De vier datum types gebruiken de **allOf** element om de basis type te overerven
```

Comment:
Misschien bij het basis type (tussen haakjes) aangeven dat dat de AbstractDatum is.

Reply:
done